File: AndroidAuth.MvvmCross/Xamarin.Auth.Droid.Tests/AccountStoreInstanceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Xamarin.Auth.Interfaces;
using Cirrious.CrossCore;
using Xamarin.Auth.Droid;

namespace KeyStoreBug.DemoApp.Tests
{
    public class AccountStoreInstanceProvider
    {
        private static List<IAccountStore> _instances = new List<IAccountStore>();

        private static string _serviceId = "DemoApp";
        
        public static IAccountStore GetNew()
        {
            var instance = new AccountStore(Mvx.Resolve<Cirrious.CrossCore.Droid.IMvxAndroidGlobals>().ApplicationContext);
            _instances.Add(instance);

            return instance;
        }

        public static void ClearAllAccounts()
        {
            _instances.ForEach(i => i.FindAccountsForService(_serviceId).ToList().ForEach(a => i.Delete(a, _serviceId)));
        }
    }
}

File: AndroiAuthNative/AndroidAuthNativeTests/src/com/xamarin/auth/testutils/AccountStoreInstanceProvider.java
//
// Translated by CS2J (http://www.cs2j.com): 02/03/2014 00:38:03
//

package com.xamarin.auth.testutils;


import android.content.Context;
import com.xamarin.auth.Account;
import com.xamarin.auth.droid.AccountStore;
import com.xamarin.auth.interfaces.IAccountStore;

import java.util.ArrayList;
import java.util.List;

public class AccountStoreInstanceProvider
{
    private static List<IAccountStore> _instances = new ArrayList<IAccountStore>();
    private static String _serviceId = "kanjigrand";
    public static IAccountStore getNew(Context ctx) throws Exception {
        IAccountStore instance = new AccountStore(ctx/*Mvx.<Cirrious.CrossCore.Droid.IMvxAndroidGlobals>Resolve().ApplicationContext*/);
        _instances.add(instance);
        return instance;
    }

    public static void clearAllAccounts() /*throws Exception*/ {
        //_instances.ForEach(i => i.FindAccountsForService(_serviceId).ToList().ForEach(a => i.Delete(a, _serviceId)));
        for(IAccountStore store : _instances)
        {
            for (Account acc : store.FindAccountsForService(_serviceId))
            {
                store.Delete(acc, _serviceId);
            }
        }
    }

}


